lsp-mode.el

```diff
@@ -4868,18 +4869,53 @@ When language is nil render as markup if `markdown-mode' is loaded."
                  (format "%s (%s)" element count))
           (push element elements))))))
 
+(defface lsp-preferred-code-action-face '((t :underline t))
+  "Face used to highlight preferred code actions.
+Used by `lsp-execute-code-action' and
+`lsp-execute-code-action-by-kind' if they have to prompt for an
+action.")
+
+(defface lsp-disabled-code-action-face '((t :inherit shadow))
+  "Face used to highlight disabled code actions.
+See `lsp-preferred-code-action-face' for details.")
+
+(defface lsp-disabled-code-action-reason-face '((t :inherit lsp-details-face))
+  "Face used to highlight the reason a code action is disabled.
+Shown after the code action in `lsp-execute-code-action',
+... (see `lsp-preferred-code-action-face').")
+
+(lsp-defun lsp--code-action-title ((action &as &CodeAction :title :is-preferred? :disabled?))
+  ;; It would be strange for a code action to be both preferred and disabled, so
+  ;; `cond' is probably good enough.
+  (cond (is-preferred?
```

Comment:
Ideally we would want the default code action to be preselected in the select dialog. If you don't want to adress it in this PR you may leave a TODO.

Reply:
I have implemented this. That feature was surprisingly easy to implement, requiring only a small change to `lsp-completing-read` and one to `lsp--select-action`.